### ISSUE TITLE
Macro resolution error

### ISSUE THREAD

**Issue by dedmen** [SQFvm/vm#60]
```
#define DOUBLES(var1,var2) var1##_##var2
#define PREFIX ace
#define COMPONENT medical_engine
#define ADDON DOUBLES(PREFIX,COMPONENT)
#define GVAR(var1) DOUBLES(ADDON,var1)
#define QUOTE(x) #x
#define QGVAR(var1) QUOTE(GVAR(var1))
#define HIT_STRUCTURAL QGVAR($#structural)
[HIT_STRUCTURAL, 0]
```

Actual:
`["ace_medical_engine_$"tructural"", 0]`
```
dical_engine_$"tructural"", 0]
               ^^^^^^^^^
[ERR][L9|C23|P:\x64\__commandlinefeed.sqf]      Expected ']'.
dical_engine_$"tructural"", 0]
               ^^^^^^^^^
[ERR][L9|C23|P:\x64\__commandlinefeed.sqf]      Expected either ';' or ','.
ine_$"tructural"", 0]
               ^
[ERR][L9|C32|P:\x64\__commandlinefeed.sqf]      Expected either ';' or ','.
Executing...
-------------------------------------------------------------------------------
-------------------------------------------------------------------------------
[WORK]  <EMPTY>
```

Expected:
`["ace_medical_engine_$#structural", 0]`

**Comment by X39** [comment]
How is `ace_medical_engine_$#structural` expected?
`$#structural` should resolove into `$structural`

**Comment by dedmen** [comment]
Maybe # only evaluates argument if it's actually macro parameter or a previously #define'd value?

**Comment by dedmen** [comment]
Config.cpp with
```
#define DOUBLES(var1,var2) var1##_##var2
#define PREFIX ace
#define COMPONENT medical_engine
#define ADDON DOUBLES(PREFIX,COMPONENT)
#define GVAR(var1) DOUBLES(ADDON,var1)
#define QUOTE(x) #x
#define QGVAR(var1) QUOTE(GVAR(var1))
#define HIT_STRUCTURAL QGVAR($#structural)
test = HIT_STRUCTURAL;
```
Into Arma 3 tools CfgConvert to binarize it.

![notepad++_2019-04-19_14-50-47](https://user-images.githubusercontent.com/3768165/56424948-85342a80-62b2-11e9-8515-6fc26020ebb4.png)


Other test
```
test = #test;
```
Result:
![notepad++_2019-04-19_14-51-48](https://user-images.githubusercontent.com/3768165/56424979-a5fc8000-62b2-11e9-8664-06723cbc5545.png)

So yes. As I said. # only stringifies if it's a macro parameter, or a #define'd value.

**Comment by X39** [comment]
Thats a negative
in macro parameters, `#` seems to never stringify a value:
```
#define TEST_STR3_(A) #A
#define TEST_STR3(A) TEST_STR3_(#A)
#define TEST_STR3_EXEC TEST_STR3(abc)
TEST_STR3_EXEC```
--> `"#abc"`

**Comment by dedmen** [comment]
Only if the macro parameter is not used to call another macro then.

**Comment by dedmen** [comment]
The off by one is also a problem

`#define PARSE(value) (call compile format ["%1", value])`
https://github.com/CBATeam/CBA_A3/blob/master/addons/optics/fnc_updateOpticInfo.sqf#L50

->

```
compile I:/CBA_A3/addons/optics/XEH_preInit.sqfc
at ["%1", _x])#, _x#1, (call c
               ^
[ERR][L50|C82|I:/CBA_A3/addons/optics/fnc_updateOpticInfo.sqf]  Missing RARG for binary operator.
at ["%1", _x])#, _x param [3,
               ^
[ERR][L50|C121|I:/CBA_A3/addons/optics/fnc_updateOpticInfo.sqf] Missing RARG for binary operator.
```

**Comment by X39** [comment]
Fixed in 9a198e9d573c98591d52fbfeb2f7f2db48a88552
